packages/rspack/src/builtin-plugin/css-extract/loader.ts

```diff
@@ -43,22 +43,32 @@ function hotLoader(
 		options: CssExtractRspackLoaderOptions;
 		locals: Record<string, string>;
 	}
-) {
-	const accept = context.locals
-		? ""
-		: "module.hot.accept(undefined, cssReload);";
+): string {
+	const localsJsonString = JSON.stringify(JSON.stringify(context.locals));
 	return `${content}
     if(module.hot) {
-      // ${Date.now()}
-      var cssReload = require(${stringifyRequest(
-				context.loaderContext,
-				path.join(__dirname, "./hmr/hotModuleReplacement.js")
-			)}).cssReload(module.id, ${JSON.stringify({
-				...context.options,
-				locals: !!context.locals
-			})});
-      module.hot.dispose(cssReload);
-      ${accept}
+      (function() {
+        var localsJsonString = ${localsJsonString};
+        // ${Date.now()}
+        var cssReload = require(${stringifyRequest(
+					context.loaderContext,
+					path.join(__dirname, "hmr/hotModuleReplacement.js")
+				)})(module.id, ${JSON.stringify(context.options)});
```

Comment:
Should this be `require().cssReload(module.id, ${JSON.stringify(context.options ?? {})`?

1. We are using ESM with `cssReload` as named export in `hotModuleReplacement.js`
2. `context.options` may be `undefined`, and we have `if (options.locals)` in `cssReload`